### PR TITLE
Add react-textarea-autosize@4.3.0 w/ npm auto-update

### LINKED ISSUE #9832: [Request] Add react-textarea-autosize
**Library name:** react-textarea-autosize
**Git repository url:** https://github.com/andreypopp/react-textarea-autosize
**npm package url(optional):** https://www.npmjs.com/package/react-textarea-autosize
**License(s):** The MIT License (MIT)
**Official homepage:** http://andreypopp.github.io/react-textarea-autosize/
**Wanna say something? Leave message here:**
Thanks a million! 👍 

=====================
Notes from cdnjs maintainer:
You are welcome to add a library via sending pull request,
it'll be faster then just opening a request issue,
and please don't forget to read the guidelines for contributing, thanks!!

### PR DESCRIPTION
**Pull request by sufuf3**
close #9832, cc @andreypopp
@kennynaoh please help me review this PR, thank you.

Pull request for issue: #9832
Related issue(s): # #

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/andreypopp/react-textarea-autosize
 * Official website (optional, not the repository): http://andreypopp.github.io/react-textarea-autosize/
 * NPM package url (optional): https://www.npmjs.com/package/react-textarea-autosize
 * License and its reference: MIT
 * GitHub / Bitbucket popularity (required):
   - Count of stars: 424
   - Count of watchers: 11
   - Count of forks: 115
 * NPM download stats (optional):
   - Downloads in the last day: 3614
   - Downloads in the last week: 19912
   - Downloads in the last month: 69120

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [x] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [x] More than 500 downloads stats per month on npm registry
 * [ ] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

File: ajax/libs/react-textarea-autosize/package.json [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    "url": "git+https://github.com/andreypopp/react-textarea-autosize.git"
+  },
+  "license": "MIT",
+  "author": {
+    "name": "Andrey Popp",
+    "email": "8mayday@gmail.com",
+    "url": "https://andreypopp.com/"
+  },
+  "filename": "TextareaAutosize.min.js",
+  "homepage": "http://andreypopp.github.io/react-textarea-autosize/",
```
Comment (kennynaoh) [inline ajax/libs/react-textarea-autosize/package.json]: It's better to use `https` here

File: ajax/libs/react-textarea-autosize/package.json [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  "npmName": "react-textarea-autosize",
+  "npmFileMap": [
+    {
+      "basePath": "lib",
+      "files": [
+        "**/TextareaAutosize.*"
```
Comment (kennynaoh) [inline ajax/libs/react-textarea-autosize/package.json]: I think `TextareaAutosize*` is better.

**Comment by kennynaoh** [comment]
The commit message is too long, please move `npm auto-update` to second line.

**Comment by sufuf3** [comment]
@kennynaoh please help me review this PR again, thank you.

**Comment by kennynaoh** [comment]
You can remove `w/` in the first line.

File: ajax/libs/react-textarea-autosize/package.json [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    "url": "https://andreypopp.com/"
+  },
+  "filename": "TextareaAutosize.min.js",
+  "homepage": "https://andreypopp.github.io/react-textarea-autosize/",
+  "npmName": "react-textarea-autosize",
+  "npmFileMap": [
+    {
+      "basePath": "lib",
+      "files": [
+        "TextareaAutosize.*"
```
Comment (kennynaoh) [inline ajax/libs/react-textarea-autosize/package.json]: And also remove the dot `.`

**Comment by sufuf3** [comment]
@kennynaoh please help me review this PR again, thank you.

**Comment by PeterDaveHello** [comment]
Looks like this PR is missed ... @sufuf3 please rebase this branch with the latest master branch, thanks.

**Comment by sufuf3** [comment]
@PeterDaveHello , I've rebased this branch done.

**Comment by PeterDaveHello** [comment]
@sufuf3 nice! @pvnr0082t can you take a look? Thanks

File: ajax/libs/react-textarea-autosize/4.0.5/TextareaAutosize.js [hunk 4]
```diff
@@ -0,0 +1,285 @@
...
+
+var _class, _temp; /**
+                    * <TextareaAutosize />
+                    */
+
+var _react = require('react');
+
+var _react2 = _interopRequireDefault(_react);
+
+var _calculateNodeHeight = require('./calculateNodeHeight');
```
Comment (pvnr0082t) [inline ajax/libs/react-textarea-autosize/4.0.5/TextareaAutosize.js]: @sufuf3 I think we need `calculateNodeHeight.js` also.

**Comment by sufuf3** [comment]
@pvnr0082t , I added calculateNodeHeight.js. please help me review this PR, thank you.

File: ajax/libs/react-textarea-autosize/package.json [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    "url": "https://andreypopp.com/"
+  },
+  "filename": "TextareaAutosize.min.js",
+  "homepage": "https://andreypopp.github.io/react-textarea-autosize/",
+  "npmName": "react-textarea-autosize",
+  "npmFileMap": [
+    {
+      "basePath": "lib",
+      "files": [
+        "**/*"
```
Comment (pvnr0082t) [inline ajax/libs/react-textarea-autosize/package.json]: It's a little bit not safe or no sense to use `**/*` under a non-dist folder.

**Comment by sufuf3** [comment]
@pvnr0082t , I added calculateNodeHeight.js. please help me review this PR again, thank you.

**Comment by PeterDaveHello** [comment]
@jdalton sorry about that.
@sufuf3 why didn't you mention the repo owner? BTW, are you sure about we should include all the files under `lib`?

**Comment by sufuf3** [comment]
@kennynaoh I modified it, please help me review this PR.
@PeterDaveHello  `TextareaAutosize.js` require('./calculateNodeHeight') , so we need also need `calculateNodeHeight.js` .

File: ajax/libs/react-textarea-autosize/package.json [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+  ],
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/andreypopp/react-textarea-autosize.git"
+  },
+  "license": "MIT",
+  "author": {
+    "name": "Andrey Popp",
+    "email": "8mayday@gmail.com",
+    "url": "https://andreypopp.com/"
```
Comment (kennynaoh) [inline ajax/libs/react-textarea-autosize/package.json]: wrong url

File: ajax/libs/react-textarea-autosize/package.json [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+  },
+  "homepage": "https://andreypopp.github.io/react-textarea-autosize/",
+  "npmName": "react-textarea-autosize",
+  "npmFileMap": [
+    {
+      "basePath": "lib",
+      "files": [
+        "+(calculateNodeHeight|TextareaAutosize)*.js"
```
Comment (kennynaoh) [inline ajax/libs/react-textarea-autosize/package.json]: I think we just use `+(calculateNodeHeight|TextareaAutosize)*`

**Comment by sufuf3** [comment]
@kennynaoh  , I modified it. please help me review this PR again, thank you.

**Comment by PeterDaveHello** [comment]
I wonder if these two files need to be bundled or browserified? Any reference of its usage that says it directly runs in browser?

**Comment by sufuf3** [comment]
in source code.
https://github.com/andreypopp/react-textarea-autosize/blob/master/src/TextareaAutosize.js#L6

**Comment by PeterDaveHello** [comment]
@sufuf3 I don't think this answered the question ...

**Comment by sufuf3** [comment]
@PeterDaveHello I didn't find the reference. So should I remove `calculateNodeHeight.js`?

**Comment by PeterDaveHello** [comment]
@sufuf3 Then how do you know that we don't need it?

**Comment by PeterDaveHello** [comment]
Any updates here? Thanks!

**Comment by sufuf3** [comment]
@PeterDaveHello I am waiting for a response from andreypopp/react-textarea-autosize#132
But no one answer me.

**Comment by sufuf3** [comment]
I discussed with the author, but there is no response from the author. 
I think the file `calculateNodeHeight.js` should include, because `TextareaAutosize.js` import `calculateNodeHeight` from `calculateNodeHeight.js`. ref: https://github.com/andreypopp/react-textarea-autosize/blob/master/src/TextareaAutosize.js#L6
So should I remove `calculateNodeHeight.js`?
Or just rebase master?

**Comment by PeterDaveHello** [comment]
So how do you know that we need it? Tested?

**Comment by andreypopp** [comment]
Sorry for not responding earlier. Yes you need code from both files for
component to work. But you have to combine those both files into a single
one somehow (producing a UMD build maybe). This library isn't distributed
in that format unfortunately.

**Comment by sufuf3** [comment]
@andreypopp thank you.
@PeterDaveHello as the author said, we should include`calculateNodeHeight.js` and `TextareaAutosize.js`. Should I rebase this PR? Or what is the next thing I should do?
Thank you.

**Comment by PeterDaveHello** [comment]
@sufuf3 are you sure about you understand that?
>  But you have to combine those both files into a single one somehow (producing a UMD build maybe). This library isn't distributed in that format unfortunately.

**Comment by sufuf3** [comment]
@PeterDaveHello 
Combine `calculateNodeHeight.js` and `TextareaAutosize.js` to a single file, like https://andreypopp.github.io/react-textarea-autosize/bundle.js . 
It seems the method as https://github.com/andreypopp/react-textarea-autosize/blob/master/example/Makefile#L20-L21 .
So could you please let me know that how should I add this library?
Thanks.

**Comment by PeterDaveHello** [comment]
We don't build files from src by our own, so should ask its author can provide that.

**Comment by sufuf3** [comment]
@andreypopp Although this library isn't distributed a single file, would you like to provide it?

**Comment by Andarist** [comment]
ive recently added UMD build - available [here](https://github.com/andreypopp/react-textarea-autosize/blob/master/dist/react-textarea-autosize.min.js)

@sufuf3 feel free to update this PR

**Comment by PeterDaveHello** [comment]
@Andarist Thanks a lot!

**Comment by sufuf3** [comment]
@Andarist Thank you very much.
I'll update this PR. :+1:

**Comment by sufuf3** [comment]
@cdnjs/intern3 Please help me review this PR, thanks.

**Comment by sufuf3** [comment]
@kennynaoh 
Please help me review this PR, Thank you.

**Comment by sufuf3** [comment]
@cdnjs/intern3 Please help me review this PR, thanks.

**Comment by sufuf3** [comment]
ping @extend1994 @hare1039 @clairetsai818

File: ajax/libs/react-textarea-autosize/package.json [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/andreypopp/react-textarea-autosize.git"
+  },
+  "license": "MIT",
+  "author": {
+    "name": "Andrey Popp",
+    "email": "8mayday@gmail.com"
+  },
+  "homepage": "https://andreypopp.github.io/react-textarea-autosize/",
```
Comment (sufuf3) [inline ajax/libs/react-textarea-autosize/package.json]: @extend1994 as @PeterDaveHello said in https://github.com/cdnjs/cdnjs/pull/11085#issuecomment-297126271
> If it at least links to its GitHub repo with a demo, I'll treat it as homepage @cdnjs/intern3

I think we can this link as the homepage.

**Comment by sufuf3** [comment]
@extend1994 @hare1039 @clairetsai818  
Please help me review this PR, Thank you.

**Comment by maruilian11** [comment]
@pvnr0082t Please help me make the double check, thanks!

**Comment by PeterDaveHello** [comment]
@sufuf3 didn't I say you should got 2 approvals from @cdnjs/intern3  before asking more reviews from mentors?

**Comment by PeterDaveHello** [comment]
@sufuf3 try to update the the newer version next time, v4.3.0 is too old the the latest release.

**Pull request merged**